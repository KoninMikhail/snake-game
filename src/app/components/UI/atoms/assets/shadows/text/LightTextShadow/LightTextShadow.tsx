import {cloneElement} from 'react';
import {IShadow} from '@ui/atoms/assets/shadows/Shadow.interface';

/**
 * @name LightTextShadow
 *
 * Use for creating text shadow.
 *
 * @param children - a Text component to add shadow
 */
export const LightTextShadow = ({ children }: IShadow) => {
    return cloneElement(children, {
        style: {
            textShadow: '2px 2px 0 rgba(0, 0, 0, 1)',
        },
    });
};
