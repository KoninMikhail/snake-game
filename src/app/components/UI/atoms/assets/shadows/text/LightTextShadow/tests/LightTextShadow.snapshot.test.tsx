import { render } from '@testing-library/react';
import { LightTextShadow } from '@ui/atoms/assets/shadows/text/LightTextShadow/LightTextShadow';

describe('LightTextShadow', () => {
    it('equal snapshot', () => {
        const { container } = render(
            <LightTextShadow>
                <div>{'Test ITextProps'}</div>
            </LightTextShadow>
        );

        expect(container).toMatchSnapshot();
    });
});
