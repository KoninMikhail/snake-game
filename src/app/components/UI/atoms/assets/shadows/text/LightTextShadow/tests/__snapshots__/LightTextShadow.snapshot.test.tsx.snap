// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LightTextShadow equal snapshot 1`] = `
<div>
  <div
    style="text-shadow: 2px 2px 0 rgba(0, 0, 0, 1);"
  >
    Test ITextProps
  </div>
</div>
`;
